src/main/java/core/basesyntax/model/Fruit.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Fruit {
+    private Map<String,Integer> fruitMap;
+
+    public Fruit() {
+        this.fruitMap = new HashMap<>();
+    }
+
+    public int getCurrentAmount(String fruitType) {
+        return fruitMap.getOrDefault(fruitType, 0);
+    }
+
+    public int balance(String fruitType, int amount) {

```

Comment:
you should implement strategy pattern, it's better to create OperationHandler, and BalanceOperationHandler etc for each operation